yarn-project/aztec.js/src/utils/portal_manager.ts

```diff
@@ -241,15 +263,22 @@ export class L1ToL2TokenPortalManager {
     const [claimSecret, claimSecretHash] = await this.bridgeSetup(amount, mint);
 
     this.logger.info('Sending L1 tokens to L2 to be claimed publicly');
-    const { request } = await this.portal.simulate.depositToAztecPublic([
-      to.toString(),
-      amount,
-      claimSecretHash.toString(),
-    ]);
-
-    const txReceipt = await this.publicClient.waitForTransactionReceipt({
-      hash: await this.walletClient.writeContract(request),
-    });
+    const args = [to.toString(), amount, claimSecretHash.toString()] as const;
+    const { request } = await this.portal.simulate.depositToAztecPublic(args);
+
+    const txReceipt = await this.l1TxUtils.sendAndMonitorTransaction(
+      {
+        to: request.address!,
+        data: encodeFunctionData({
+          abi: this.portal.abi,
+          functionName: 'depositToAztecPublic',
+          args,
+        }),
+      },
+      {
+        fixedGas: request.gas,
+      },
+    );
```

Comment:
Do we need to call `simulate` here if the tx utils already call `estimateGas` under the hood, which should throw (emphasis on should, we need to verify that) if the tx would revert? Also, note that `simulate` does not seem to perform a gas estimation as it's just mapped to `call`, and the `gas` returned seems to be whatever was originally set and not the actual gas used (see [here](https://github.com/wevm/viem/blob/591f50c59f6bb0172d1bd3a77a50c04cae56d636/src/actions/public/simulateContract.ts#L293)).